app/controllers/publishers/site_banners_controller.rb

```diff
@@ -39,32 +40,53 @@ def update_background_image
     end
 
     site_banner = current_publisher.site_banner
-    update_image(site_banner.background_image)
+    update_image(attachment: site_banner.background_image, attachment_type: SiteBanner::BACKGROUND)
     head :ok
   end
 
   private
 
-  def update_image(attachment)
+  def update_image(attachment:, attachment_type:)
     data_url = params[:image].split(',')[0]
     if data_url.starts_with?("data:image/jpeg")
       content_type = "image/jpeg"
       extension = ".jpg"
     elsif data_url.starts_with?("data:image/png")
       content_type = "image/png"
       extension = ".png"
+    elsif data_url.starts_with?("data:image/bmp")
+      content_type = "image/bmp"
+      extension = ".bmp"
     else
-      # TODO: Throw an exception here
+      LogException.perform(StandardError.new("Unknown image format:" + data_url), params: {})
+      return nil
     end
-    filename = Time.now.to_s.gsub!(" ", "_").gsub!(":", "_") + current_publisher.id + "_logo"
+    filename = Time.now.to_s.gsub!(" ", "_").gsub!(":", "_") + current_publisher.id
 
-    file = Tempfile.new([filename, extension])
-    File.open(file.path, 'wb') do |f|
+    temp_file = Tempfile.new([filename, extension])
+    File.open(temp_file.path, 'wb') do |f|
```

Comment:
should we explicitly close this tempfile?

Reply:
Per https://ruby-doc.org/core-2.5.1/File.html#M000069 and https://stackoverflow.com/questions/4795447/rubys-file-open-and-the-need-for-f-close

`If the optional code block is given, it will be passed the opened file as an argument and the File object will automatically be closed when the block terminates.`